Address issue #1753, fix FloatCanvas GUIMode GUIZoomIn and GUIZoomOut…

DKWoods opened the pull request:
Fixes #1753

At some point, FloatCanvas' Zoom method changed the values expected in the centerCoords parameter from "world" or "pixel" to "World" or "Pixel".  The parallel change was not made in the GUIMode's GUIZoomIn and GUIZoomOut objects, causing them not to function properly.  This PR fixes the case of that parameter in those objects.

RobinD42 commented:
It looks like there are also some changes needed in the `FloatCanvas.Zoom` method's docstring, which shows the lowercase values are the accepted options.

OTOH, @swt2c's [suggestion](https://github.com/wxWidgets/Phoenix/issues/1753#issuecomment-665150001) to do case-insensitive comparisons. That will take care of people still using the Uppercased values, and will not break things again for people that have already switched to the lowercase values.

DKWoods commented:
Made FloatCanvas.Zoom method's centerCoords parameter case insensitive.  Added exception handling for unknown parameter value parallel with other FloatCanvas methods.  Also made FloatCanvas.MoveImage method's CoordType parameter case insensitive to improve consistency.  Fixed docstring to InitialCaps for consistency with other FloatCanvas methods.

DKWoods commented:
I fixed the docstring and made the Zoom method case insensitive. I 
noticed that the MoveImage method had a similar parameter that required 
Iniital-Caps, so made that case insensitive as well. MoveImage threw a 
FloatCanvasError exception for an unrecognized parameter, so I added 
that to the Zoom method as well.

If I understand Git Pull Requests correctly, all I needed to do was to 
commit these changes and push them to my fork, and they'll automatically 
get added to the existing Pull Request.  If I'm wrong about that, please 
let me know.

David

> It looks like there are also some changes needed in the 
> |FloatCanvas.Zoom| method's docstring, which shows the lowercase 
> values are the accepted options.
>
> OTOH, @swt2c <https://github.com/swt2c>'s suggestion 
> <https://github.com/wxWidgets/Phoenix/issues/1753#issuecomment-665150001> 
> to do case-insensitive comparisons. That will take care of people 
> still using the Uppercased values, and will not break things again for 
> people that have already switched to the lowercase values.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/wxWidgets/Phoenix/pull/1754#issuecomment-665949892>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AC7ECDLABULBW5PNO7A6W4LR6CLDPANCNFSM4PJKCXCQ>.
>
-- 
David Woods
Pronouns:  he, him, his

swt2c commented:
On Wed, 29 Jul 2020, David Woods wrote:

> If I understand Git Pull Requests correctly, all I needed to do was to
> commit these changes and push them to my fork, and they'll automatically
> get added to the existing Pull Request.  If I'm wrong about that, please
> let me know.

Yes, that's all you need to do (and you can see your new commit in the 
pull request).  My personal preference though, when revising a PR, is to 
amend my previous commit(s) rather than adding new ones.

Scott
